src/sentry/services/hybrid_cloud/email/__init__.py

```diff
@@ -0,0 +1,174 @@
+from __future__ import annotations
+
+import abc
+from dataclasses import dataclass
+from typing import Collection, Iterable, Type
+
+from sentry.services.hybrid_cloud import InterfaceWithLifecycle, silo_mode_delegation, stubbed
+from sentry.services.hybrid_cloud.organization import ApiOrganization, ApiOrganizationMember
+from sentry.services.hybrid_cloud.user import APIUser
+from sentry.silo import SiloMode
+from sentry.utils import metrics
+
+
+@dataclass
+class ApiUserEmail:
+    id: int
+    user: APIUser
+    email: str
+    is_verified: bool
+    is_primary: bool
+
+
+class AmbiguousUserFromEmail(Exception):
+    def __init__(self, email: str, users: Collection[APIUser]) -> None:
+        super().__init__(f"Resolved {email!r} to {[user.id for user in users]}")
+        self.email = email
+        self.users = tuple(users)
+
+
+class EmailService(InterfaceWithLifecycle):
+    def resolve_email_to_user(
+        self, email: str, organization: ApiOrganization | None = None
+    ) -> APIUser | None:
```

Comment:
I'm experimentally trying something a little different with this service interface, as previously discussed with @corps.

With the other HC services, the set of abstract methods are also the public contract. Here, `resolve_email_to_user` is the only method we need to expose in the public contract; the two abstract methods are implementation details that it uses. That is, in Java-speak, `get_user_emails` and `get_members_for_users` could be considered "protected abstract" methods, while `resolve_email_to_user` is "public final".